File: app/src/main/java/com/persona5dex/repositories/MainPersonaRoomRepository.java
package com.persona5dex.repositories;

import androidx.lifecycle.LiveData;

import com.persona5dex.models.MainListPersona;
import com.persona5dex.models.room.PersonaDatabase;

import java.util.List;

import javax.inject.Inject;

/**
 * Created by Rechee on 11/18/2017.
 */

public class MainPersonaRoomRepository implements MainPersonaRepository {

    private final PersonaDatabase db;

    @Inject
    public MainPersonaRoomRepository(PersonaDatabase personaDatabase){
        this.db = personaDatabase;
    }

    @Override
    public LiveData<List<MainListPersona>> getAllPersonasForMainList() {
        return db.personaDao().getAllPersonasForMainListLiveData();
    }

    @Override
    public LiveData<List<MainListPersona>> getDLCPersonas() {
        return db.personaDao().getDLCPersonas();
    }

    @Override
    public LiveData<String> getPersonaName(int personaID) {
        return db.personaDao().getPersonaName(personaID);
    }
}

File: app/src/main/java/com/persona5dex/repositories/MainPersonaRoomRepository.kt
package com.persona5dex.repositories

import androidx.lifecycle.LiveData
import com.persona5dex.models.MainListPersona
import com.persona5dex.models.room.PersonaDatabase
import javax.inject.Inject

/**
 * Created by Rechee on 11/18/2017.
 */
class MainPersonaRoomRepository @Inject constructor(private val db: PersonaDatabase) : MainPersonaRepository {
    override val allPersonasForMainListLiveData: LiveData<List<MainListPersona>>
        get() = db.personaDao().allPersonasForMainListLiveData

    override val allPersonasForMainList: List<MainListPersona>
        get() = db.personaDao().allPersonasForMainList

    override val dLCPersonas: LiveData<List<MainListPersona>>
        get() = db.personaDao().dLCPersonas

    override fun getPersonaName(personaID: Int): LiveData<String?> {
        return db.personaDao().getPersonaName(personaID)
    }
}